account_invoice_base_invoicing_mode/models/sale_order.py

```diff
@@ -62,3 +62,14 @@ def generate_invoices_by_invoice_mode(
             self.with_delay()._generate_invoices_by_partner(saleorder_ids)
         companies.write({last_execution_field_name: datetime.now()})
         return saleorder_groups
+
+    def _create_invoices(self, grouped=False, final=False, date=None):
+        moves = self.env["account.move"]
+        for partner_invoice in self.partner_invoice_id:
```

Comment:
this breaks if you work on multiple sale orders

Reply:
I tried with multiple sale orders and multiple customers, i had no problem, what is the breaking case ?

Reply:
any code of the form (saleorder1 + saleorder2)._create_invoices() will break here. Core supports this, so should you

Reply:
In the test of this module, I tried `(self.so1 + self.so2)._create_invoices()`without any error.

Reply:
oh apologies, right, I was confused with scalar fields. So what you do here is group sale orders by partner_invoice_id, and pass this partner's choice of grouping. Must have been tired yesterday.

This will not apply to deposits, but then I also don't see so much point in grouping deposits.